src/p2sh.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * This module provides functions and constants for the P2SH address type.
+ */
+
+/**
+ * Address type constant for "pay-to-script-hash" (P2SH) addresses.
+ */
+export const P2SH = "P2SH";
```

Comment:
I guess this is something you were trying to fix too, like Network where we could have `AddressType.P2SH`

Reply:
Yes. A change on this was attempted, but the scope is greater than Networks. While chasing it all down, I didn't complete it, and I left it out. It could still be added in if you feel it's worth it.